src/bin/avail-light.rs

```diff
@@ -335,32 +344,40 @@ async fn run(error_sender: Sender<anyhow::Error>) -> Result<()> {
 		state.clone(),
 	));
 
-	let (message_tx, message_rx) = channel::<(Header, Instant)>(128);
-
-	tokio::task::spawn(avail_light::subscriptions::finalized_headers(
-		rpc_client.clone(),
-		message_tx,
-		error_sender.clone(),
-		state.clone(),
-		db.clone(),
-	));
+	let (message_tx, message_rx) = broadcast::channel::<(Header, Instant)>(128);
```

Comment:
Why change from `mpsc` to `broadcast` here? "Who" else is receiving from the `message_rx`?

Reply:
One is the light client which starts with the random sampling when header is verified, and another is task that sends header verified message to connected web socket clients. I guess that confusion comes from broadcast API which creates another receiver by subscribint to the sender:

```rust
message_tx.subscribe()
```

instead of cloning the sender.